index.html

```diff
@@ -3697,149 +3786,181 @@ <h3>The <dfn>ThingFilter</dfn> dictionary</h3>
+        but these are implemented by dedicated methods now.
       </p>
-      <!--p>
-        The <dfn>query</dfn> property represents a query string accepted by the implementation, for instance a SPARQL or JSON query. Support may be implemented locally in the <a>WoT Runtime</a> or remotely as a service in a <a>Thing Directory</a>.
-      </p-->
     </section>
 
     <section>
-      <h3>The <dfn>start()</dfn> method</h3>
+      <h3>The <dfn data-lt="generic-discovery">generic discovery</dfn> algorithm</h3>
       <div>
-        Starts the discovery process. The method MUST run the following steps:
+        Starts a generic discovery process by any means supported by the underlying
+        implementation, for instance by using multicast, preconfigured directory, etc.
+        The method, given |discovery:ThingDiscoveryProcess| and |discoveryMethod|
+        MUST run the following steps:
         <ol>
           <li>
-            If invoking this method is not allowed for the current scripting context for security reasons, set the <a href="#dom-thingdiscovery-error">error</a> property to a {{SecurityError}} and abort these steps.
+            Let |filter| denote the [[\filter]] internal slot of |discovery|.
           </li>
           <li>
-            If discovery is not supported by the implementation, set the <a href="#dom-thingdiscovery-error">error</a> property to {{NotSupportedError}} and abort these steps.
+            If |discoveryMethod| is `"generic"`, request the underlying platform
+            to start the discovery process by any means supported and
+            preconfigured in the <a>WoT Runtime</a> for which the script has
+            access to, passing |filter| to it.
           </li>
           <li>
-            Let |filter| denote the <a href="#dom-thingdiscovery-filter">filter</a> property.
-          </li>
-          <!--li>
-            If the |filter| is defined,
-            <ul>
+            Otherwise if |discoveryMethod| is `"directory"`,
+            <ol>
               <li>
-                If |filter|'s <a href="#dom-thingfilter-query">|query|</a> is defined, pass it as an opaque string to the underlying implementation to be matched against discovered items. The underlying implementation is responsible to parse it e.g. as a SPARQL or JSON query and match it against the <a>Thing Description</a>s found during the discovery process. If queries are not supported, set |this.error| to {{NotSupportedError}} and abort these steps.
+                Let |url:USVString| denote the [[\url]] <a>internal slot</a>
+                of |discovery|.
               </li>
-            </ul>
-          </li-->
-          <!-- <li>
-            Create the <a>discovery results</a> <a>internal slot</a> for storing discovered {{ThingDescription}} objects.
-          </li> -->
+              <li>
+                Request the underlying platform to start the directory discovery
+                passing |url| and |filter| to it.
```

Comment:
Is this enough, or do we need to be more specific here? Given that a directory is described by a Thing Description (derived from [this Thing Model](https://w3c.github.io/wot-discovery/#directory-api-spec)), I assume we could also specify that this TD needs to be consumed and the `things` property needs to be read?

Reply:
We discussed in the call that we might need an algorithm here, imported/adapted from the Discovery spec. 
However, if there are differences in protocol bindings, it is not clear where these algorithms need to be specified: in the Discovery spec, or in Bindings, or in Scripting. 
It feels awkward to specify in detail what to do with the content of a TD... that has not been in the scope of Scripting API.
This is one thing to figure out in this PR - for now, leaving it open for comments.